src/Microsoft.Azure.WebJobs.Host/Listeners/FunctionListener.cs

```diff
@@ -54,7 +57,54 @@ public void Dispose()
             }
             catch (Exception e)
             {
-                new FunctionListenerException(_descriptor.ShortName, e).TryRecover(_trace, _logger);
+                var flx = new FunctionListenerException(_descriptor.ShortName, e);
+                if (_allowPartialHostStartup)
+                {
+                    flx.Handled = true;
+                }
+
+                flx.TryRecover(_trace, _logger);
+
+                if (_allowPartialHostStartup)
+                {
+                    // if to here, the listener exception was handled and
+                    // we're in partial startup mode.
+                    // we spin up a background task retrying to start the listener
+                    Task taskIgnore = Task.Run(() => RetryStartWithBackoffAsync(cancellationToken), cancellationToken);
```

Comment:
Just curious, what does the additional Task.Run(..) achieve here? How is it different from just 
`Task taskIgnore = RetryStartWithBackoffAsync(cancellationToken);

Reply:
My understanding is that invoking directly (fire and forget) will capture the current synchronization context and resume on that context. Task.Run will just start and complete on a thread pool thread without capturing/restoring a context.